crypto/ts/ts_rsp_sign.c

```diff
@@ -58,46 +59,27 @@ static ASN1_INTEGER *def_serial_cb(struct TS_resp_ctx *ctx, void *data)
     return NULL;
 }
 
-#if defined(OPENSSL_SYS_UNIX)
-
 static int def_time_cb(struct TS_resp_ctx *ctx, void *data,
                        long *sec, long *usec)
 {
-    struct timeval tv;
-    if (gettimeofday(&tv, NULL) != 0) {
-        ERR_raise(ERR_LIB_TS, TS_R_TIME_SYSCALL_ERROR);
-        TS_RESP_CTX_set_status_info(ctx, TS_STATUS_REJECTION,
-                                    "Time is not available.");
-        TS_RESP_CTX_add_failure_info(ctx, TS_INFO_TIME_NOT_AVAILABLE);
-        return 0;
-    }
-    *sec = tv.tv_sec;
-    *usec = tv.tv_usec;
+    OSSL_TIME t;
+    uint64_t ticks;
 
-    return 1;
-}
-
-#else
-
-static int def_time_cb(struct TS_resp_ctx *ctx, void *data,
-                       long *sec, long *usec)
-{
-    time_t t;
-    if (time(&t) == (time_t)-1) {
+    t = ossl_time_now();
+    if (ossl_time_is_zero(t)) {
         ERR_raise(ERR_LIB_TS, TS_R_TIME_SYSCALL_ERROR);
         TS_RESP_CTX_set_status_info(ctx, TS_STATUS_REJECTION,
                                     "Time is not available.");
         TS_RESP_CTX_add_failure_info(ctx, TS_INFO_TIME_NOT_AVAILABLE);
         return 0;
     }
-    *sec = (long)t;
-    *usec = 0;
+    ticks = ossl_time2ticks(t);
+    *sec = (long int)(ticks / OSSL_TIME_SECOND);
+    *usec = (long int)((ticks % OSSL_TIME_SECOND) / OSSL_TIME_US);
```

Comment:
Would it make more sense to use a `struct timeval` for this, rather than ticks?

Reply:
So long as timeval doesn't ever represent times & get an Epoch conversion, this is worthwhile.